glib/src/subclass/signal.rs

```diff
@@ -471,12 +464,22 @@ impl<'a> SignalBuilder<'a> {
     /// This is called if multiple signal handlers are connected to the signal for accumulating the
     /// return values into a single value.
     pub fn accumulator<
-        F: Fn(&SignalInvocationHint, &mut Value, &Value) -> bool + Send + Sync + 'static,
+        R: ValueType,
+        F: Fn(&SignalInvocationHint, R, &R) -> ControlFlow<R, R> + Send + Sync + 'static,
     >(
         mut self,
         func: F,
     ) -> Self {
-        self.accumulator = Some(Box::new(func));
+        self.accumulator = Some(Box::new(move |hint, accu, value| {
+            let curr_accu = accu.get::<R>().unwrap();
+            let value = value.get::<R>().unwrap();
```

Comment:
This doesn't allow getting the values by reference so would cause needless copies